newdle/client/src/components/answer/AnswerGrid.js

```diff
@@ -0,0 +1,703 @@
+    ];
+  }
+
+  return (
+    <div className={styles['answer-grid']}>
+      <Table textAlign="center">
+        <TableHeader timeslots={timeslots} interactive={false} isCreator={false} />
+        <Table.Body>
+          {participants.map(p => {
+            const selectable = participant
+              ? participant.id === p.id
+              : user
+              ? user.uid === p.auth_uid
+              : p.id === -1;
+            return (
+              <AnswerRow
+                key={p.id}
+                participant={p}
+                timeslots={timeslots}
+                answers={answers}
+                selectable={selectable}
+                hasBusyTimes={hasBusyTimes}
+                busyTimes={busyTimes}
+                duration={duration}
+                unknown={unknown && selectable}
+              />
+            );
+          })}
+        </Table.Body>
+        <TableFooter participants={participants} timeslots={timeslots} interactive={false} />
+      </Table>
+    </div>
+  );
+}
+
+AnswerGrid.propTypes = {
+  participant: PropTypes.object,
+  user: PropTypes.object,
+  name: PropTypes.string,
+  comment: PropTypes.string,
+  unknown: PropTypes.bool.isRequired,
+  hasBusyTimes: PropTypes.bool,
+  busyTimes: PropTypes.object.isRequired,
+  duration: PropTypes.number.isRequired,
+};
+
+export function ParticipantGrid({finalDate, setFinalDate, isCreator, finalized}) {
+  const timeslots = useSelector(getNewdleTimeslots);
+  const participants = useSelector(getNewdleParticipants);
+  const [hoveredColumn, setHoveredColumn] = useState(null);
+
+  if (timeslots.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className={styles['participant-grid']}>
+      <Table textAlign="center">
+        <TableHeader
+          timeslots={timeslots}
+          interactive={!finalized}
+          finalDate={finalDate}
+          setFinalDate={setFinalDate}
+          isCreator={isCreator}
+          hoveredColumn={hoveredColumn}
+          setHoveredColumn={setHoveredColumn}
+        />
+        <Table.Body>
+          {_.sortBy(participants, ['name']).map(participant => (
+            <SummaryRow
+              key={participant.id}
+              participant={participant}
+              timeslots={timeslots}
+              interactive={!finalized}
+              finalDate={finalDate}
+              hoveredColumn={hoveredColumn}
+              setHoveredColumn={setHoveredColumn}
+            />
+          ))}
+        </Table.Body>
+        <TableFooter
+          participants={participants}
+          timeslots={timeslots}
+          interactive={!finalized}
+          finalDate={finalDate}
+          hoveredColumn={hoveredColumn}
+          setHoveredColumn={setHoveredColumn}
+        />
+      </Table>
+    </div>
+  );
+}
+
+ParticipantGrid.propTypes = {
+  finalDate: PropTypes.string,
+  setFinalDate: PropTypes.func.isRequired,
+  isCreator: PropTypes.bool.isRequired,
+  finalized: PropTypes.bool.isRequired,
+};
```

Comment:
700 lines of code... maybe worth splitting this into multiple files?

Reply:
I split it into `AnswerGrid` and `ParticipantGrid` with an extra file `GridCommon` containing shared components.